fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java

```diff
@@ -320,13 +329,60 @@ public Table getTable(HiveTableName hiveTableName) {
 
     private Table loadTable(HiveTableName hiveTableName) throws TException, DdlException {
         org.apache.hadoop.hive.metastore.api.Table hiveTable = client.getTable(hiveTableName);
-        HiveTable table =  HiveMetaStoreTableUtils.convertToSRTable(hiveTable, resourceName);
+        Table table = null;
+        if (hiveTable.getSd().getInputFormat().contains("hudi")) {
+            Schema hudiTable = loadHudiTable(hiveTable);
+            table = HiveMetaStoreTableUtils.convertHudiConnTableToSRTable(hudiTable, hiveTable, resourceName);
+        } else {
+            table = HiveMetaStoreTableUtils.convertHiveConnTableToSRTable(hiveTable, resourceName);
+        }
         tableColumnStatsCache.invalidate(new HiveTableColumnsKey(hiveTableName.getDatabaseName(),
                 hiveTableName.getTableName(), null, null, table.getType()));
 
         return table;
     }
 
+    private Schema loadHudiTable(org.apache.hadoop.hive.metastore.api.Table hiveTable) throws DdlException {
+        Map<String, String> hudiProperties = Maps.newHashMap();
```

Comment:
It is better to put loadHudiTable and HudiTable.validate together

Reply:
> It is better to put loadHudiTable and HudiTable.validate together

done

Reply:
I mean loadHudiSchema and HudiTable.validate are duplicate.